pkg/sql/sem/builtins/builtins.go

```diff
@@ -8715,6 +8729,69 @@ func truncateTimestamp(fromTime time.Time, timeSpan string) (*tree.DTimestampTZ,
 	return tree.MakeDTimestampTZ(toTime, time.Microsecond)
 }
 
+func truncateInterval(fromInterval *tree.DInterval, timeSpan string) (*tree.DInterval, error) {
+
+	toInterval := tree.DInterval{}
+
+	switch timeSpan {
```

Comment:
nit: can we move strings.ToLower to here?

Reply:
The other similar truncate/extract function call sites all are doing `strings.ToLower` before calling the function.  If we do it for this one I think we should do it for all of them, but it should probably be in a separate PR.